File: spec/java/src/io/kaitai/struct/spec/TestSwitchCast.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.SwitchCast;
import org.testng.Assert;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;
import static org.testng.Assert.assertThrows;

public class TestSwitchCast extends CommonSpec {
    @Test
    public void testSwitchCast() throws Exception {
        final SwitchCast r = SwitchCast.fromFile(SRC_DIR + "switch_opcodes.bin");

        assertEquals(r.firstObj().value(), "foobar");
        assertEquals(r.secondVal().intValue(), 0x42);

        assertThrows(
                ClassCastException.class,
                new Assert.ThrowingRunnable() {
                    @Override
                    public void run() throws Throwable {
                        r.errCast();
                    }
                }
        );
    }
}

File: spec/python/spec/test_switch_cast.py
import unittest
from testformats.switch_cast import SwitchCast

class TestSwitchCast(unittest.TestCase):
    def test_switch_cast(self):
        with SwitchCast.from_file('src/switch_opcodes.bin') as r:
            self.assertEqual(r.first_obj.value, "foobar")
            self.assertEqual(r.second_val, 0x42)

            self.assertNotIsInstance(r.err_cast, SwitchCast.Strval)
            self.assertIsInstance(r.err_cast, SwitchCast.Intval)
